pkg/gateway/lb/handler/grpc/handler.go

```diff
@@ -1144,6 +1251,31 @@ func (s *server) linearsearch(ctx context.Context, cfg *payload.Search_Config,
 				}
 				return nil, err
 			}
+
+			if min != 0 && len(res.GetResults()) < min {
+				if err == nil {
+					err = errors.ErrNotEnoughSearchResult
+				}
+				if span != nil {
+					span.SetStatus(trace.StatusCodeNotFound(err.Error()))
+				}
+				err = status.WrapWithNotFound(
+					"error search result length less than minimum number",
```

Comment:
How about showing the real result length?
It is more convenient for the user.

Reply:
same
https://github.com/vdaas/vald/pull/1546#discussion_r802291404